data/emulator.js

```diff
@@ -2721,21 +2716,22 @@ class EmulatorJS {
     setupKeys(){
         for (let i=0; i<4; i++) {
             for (let j=0; j<30; j++) {
-                if (this.controls[i][j] && this.controls[i][j].value === this.keyMap[this.keyLookup(this.controls[i][j])]) {
-                    this.controls[i][j].keycode = Number(this.keyLookup(this.controls[i][j]));
+                if (this.controls[i][j] && this.keyMap) {
+                    this.controls[i][j].value = Number(this.keyLookup(this.controls[i][j]));
```

Comment:
so... value can be a keycode or a string?

Reply:
no